src/EGraphs/egraph.jl

```diff
@@ -425,21 +426,23 @@ function process_unions!(g::EGraph{ExpressionType,AnalysisType})::Int where {Exp
       eclass = g.classes[eclass_id_key]
 
       node_data = make(g, node)
-      if !isnothing(eclass.data)
-        joined_data = join(eclass.data, node_data)
-
-        if joined_data != eclass.data
-          g.classes[eclass_id_key] = EClass{AnalysisType}(eclass_id, eclass.nodes, eclass.parents, joined_data)
-          # eclass.data = joined_data
+      if !isnothing(node_data)
+        if !isnothing(eclass.data)
+          joined_data = join(eclass.data, node_data)
+        
+          if joined_data != eclass.data
+            g.classes[eclass_id_key] = EClass{AnalysisType}(eclass_id, eclass.nodes, eclass.parents, joined_data)
+            # eclass.data = joined_data
+            modify!(g, eclass)
+            append!(g.analysis_pending, eclass.parents)
+          end
+        else
+          g.classes[eclass_id_key] = EClass{AnalysisType}(eclass_id, eclass.nodes, eclass.parents, node_data)
```

Comment:
Line 440 is the same as line 439 in the original code. Only the indent was changed, because we only need to update the eclass when node_data is not nothing.

I agree that we can probably remove the commented code.

Reply:
Blame shows that @0x0f0f0f changed EClasses to immutable for the 3.0 PR. 
06c49981aafa685638ac8661d916507898480fd4

@0x0f0f0f, do you have any objections to change that back to a mutable (or partially mutable) type (if benchmarks show no effect)?

Reply:
It's ok if benchmarks show no effect

Reply:
We could address in another MR